app/helpers/my_facilities_helper.rb

```diff
@@ -18,4 +18,43 @@ def opd_load(facility, selected_period)
       facility.monthly_estimated_opd_load
     end
   end
+
+  def drug_stock_tooltip(report)
+    return nil if report.nil? || report[:patient_days] == "error"
+    tooltip_template = ERB.new <<-EOF
+      <% report[:stocks_on_hand].each_with_index do |stock_on_hand, i| %>
+        <span class='math'><%= '+' unless i == 0 %><%= stock_on_hand[:in_stock] %>*<%= stock_on_hand[:coefficient] %></span>
+        <%= stock_on_hand[:protocol_drug].name %> <%= stock_on_hand[:protocol_drug].dosage %>
+        <br>
+      <% end %>
+      <span class='math'>
+        /<%= report[:patient_count] %>*<%= report[:load_coefficient] %>*<%= report[:new_patient_coefficient] %>
+      </span>
+      Patients
+      <br>
+      <span class='math'>=<%= report[:patient_days] %></span>
+      Patient days
+    EOF
+    tooltip_template.result(binding)
```

Comment:
Is there a better way to do this where the ERB isn't in plain text? Don't want to read it from a file either.

Reply:
This is rendered a few times for every row, and we might have a few thousand rows at a time. Having gone the partial route with the form per row, we saw a lot of latency (10s of seconds) added to rendering time of the page.

In addition, this is actually a string that goes into a data field in html.

Will still give the partial a try to see if the latency impact is worth it. cc @vkrmis

Reply:
Neat. If performance sucks, then let's leave it like this and add a comment here explaining why this isn't a partial.

Reply:
We measured the performance, and it looks like partials take twice as long to render. However, the difference for about a hundred facilities is in hundreds of milliseconds, so it's okay to keep the tooltip in the partial for now. We can move it into a helper if we notice it getting slower with prod data.

Measurements in ms (page, view times):
- with partial, soft refresh: 847, 562
- with partial, hard refresh: 1219, 697
- without partial, hard refresh: 702, 356
- without partial, soft refresh: 770, 321

Reply:
Fixed in 3ea0e0ee7465eced766cfb21f091eb5960559ec6.